src/runmode-netmap.c

```diff
@@ -342,19 +341,12 @@ static void *ParseNetmapConfig(const char *iface_name)
     }
 
 #endif
-    /* netmap needs all offloading to be disabled */
-    if (aconf->in.real) {
-        char base_name[sizeof(aconf->in.iface)];
-        strlcpy(base_name, aconf->in.iface, sizeof(base_name));
-        /* for a sw_ring enabled device name, strip the trailing char */
-        if (aconf->in.sw_ring) {
-            base_name[strlen(base_name) - 1] = '\0';
-        }
-
+    /* netmap needs all hardware offloading to be disabled */
+    if (aconf->in.real && !aconf->in.sw_ring) {
         if (LiveGetOffload() == 0) {
-            (void)GetIfaceOffloading(base_name, 1, 1);
+            (void)GetIfaceOffloading(aconf->in.iface, 1, 1);
```

Comment:
why aren't we using base_name here? IIRC passing the names with netmap magic characters won't work in these ioctl wrappers.

Reply:
I concur. This patch went back and forth a few times with changes during the troubleshooting phase, and in one of those passes the code was made to mimic nearly 100% the 6.0.8 netmap code when the legacy API was used (meaning the NETMAP_V14 option was "off"). I will change this call to use base_name.